external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -423,13 +423,20 @@ private[kafka010] class KafkaOffsetReader(
 
 private[kafka010] object KafkaOffsetReader {
 
-  def kafkaSchema: StructType = StructType(Seq(
-    StructField("key", BinaryType),
-    StructField("value", BinaryType),
-    StructField("topic", StringType),
-    StructField("partition", IntegerType),
-    StructField("offset", LongType),
-    StructField("timestamp", TimestampType),
-    StructField("timestampType", IntegerType)
-  ))
+  def kafkaSchema(includeHeaders: Boolean): StructType = {
```

Comment:
nit: maybe use the following lazy creation to avoid creating the schema in every call.
```scala
  private lazy val schemaWithoutHeaders = new StructType(Array(
    StructField("key", BinaryType),
    StructField("value", BinaryType),
    StructField("topic", StringType),
    StructField("partition", IntegerType),
    StructField("offset", LongType),
    StructField("timestamp", TimestampType),
    StructField("timestampType", IntegerType)
  ))

  private lazy val schemaWithHeaders = {
    val headersType = ArrayType(StructType(Array(
      StructField("key", StringType),
      StructField("value", BinaryType))))
    new StructType(schemaWithoutHeaders.fields :+ StructField("headers", headersType))
  }

  def kafkaSchema(includeHeaders: Boolean): StructType = {
    if (includeHeaders) schemaWithHeaders else schemaWithoutHeaders
  }
```

Reply:
These don't need to be lazy. They won't be created in every call; they're in the `object`